crates/cashu/src/nuts/nut17.rs

```diff
@@ -0,0 +1,73 @@
+//! Mint via Bitcoin Onchain
+// https://github.com/cashubtc/nuts/blob/main/17.md
+use serde::{Deserialize, Serialize};
+
+use super::nut04::MintMethodSettings;
+use super::{BlindSignature, BlindedMessage, CurrencyUnit};
+use crate::types::MintQuote;
+use crate::Amount;
+
+/// Mint quote request [NUT-17]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MintQuoteBtcOnchainRequest {
+    /// Amount
+    pub amount: Amount,
+    /// Unit wallet would like to pay with
+    pub unit: CurrencyUnit,
+}
+
+/// Mint quote response [NUT-17]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MintQuoteBtcOnchainResponse {
+    /// Quote Id
+    pub quote: String,
+    /// Payment request to fulfil
+    pub address: String,
+    /// Whether the the request haas be paid
+    pub paid: bool,
+    /// Unix timestamp until the quote is valid
+    pub expiry: u64,
+}
+
+impl From<MintQuote> for MintQuoteBtcOnchainResponse {
+    fn from(mint_quote: MintQuote) -> MintQuoteBtcOnchainResponse {
+        MintQuoteBtcOnchainResponse {
+            quote: mint_quote.id,
+            address: mint_quote.request,
+            paid: mint_quote.paid,
+            expiry: mint_quote.expiry,
+        }
+    }
+}
+
+/// Mint request [NUT-17]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MintBtcOnchainRequest {
+    /// Quote id
+    pub quote: String,
+    /// Outputs
+    pub outputs: Vec<BlindedMessage>,
+}
+
+impl MintBtcOnchainRequest {
+    pub fn total_amount(&self) -> Amount {
+        self.outputs
+            .iter()
+            .map(|BlindedMessage { amount, .. }| *amount)
+            .sum()
```

Comment:
You can use `fold` to avoid mapping and summing

```suggestion
            .map(|BlindedMessage { amount, .. }| *amount)
            .sum()
```